popup.js

```diff
@@ -210,6 +210,78 @@ const setBlockExecutionStatus = (blockExecutionValue) => {
   });
 };
 
+function getTimeSinceEvent(eventTimestamp) {
+  const currentTimestamp = Date.now();
+  const timeDifference = currentTimestamp - eventTimestamp;
+
+  const seconds = Math.floor(timeDifference / 1000);
+  const minutes = Math.floor(seconds / 60);
+  const hours = Math.floor(minutes / 60);
+  const days = Math.floor(hours / 24);
+
+  let timeSince = '';
+
+  if (seconds < 60) {
+    timeSince = `${seconds}s`;
+  } else if (minutes < 60) {
+    timeSince = `${minutes}m`;
+  } else if (hours < 24) {
+    timeSince = `${hours}h`;
+  } else {
+    timeSince = `${days}d`;
+  }
+
+  return `(${timeSince})`;
+}
```

Comment:
Browsers have a native API for relative time formatting which may be useful for you here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat